src/System.IO.Compression/src/System/IO/Compression/Crc32Helper.cs

```diff
@@ -495,5 +528,16 @@ static public uint UpdateCrc32(uint crc32, byte[] buffer, int offset, int length
             crc32 ^= 0xFFFFFFFFU;
             return crc32;
         }
+        public uint crc32_zlib(uint crc32, byte[] buffer, int offset, int length)
+        {
+            Debug.Assert((buffer != null) && (offset >= 0) && (length >= 0)
+                    && (offset <= buffer.Length - length), "check the caller");
+
+            _zlibStream = new ZLibNative.ZLibStreamHandle();
```

Comment:
Why do we need to allocate a handle here?  Can't this just call the static Interop.zlib.crc32?